metric_learn/sdml.py

```diff
@@ -42,21 +42,22 @@ def __init__(self, balance_param=0.5, sparsity_param=0.01, use_cov=True,
     self.use_cov = use_cov
     self.verbose = verbose
 
-  def _prepare_inputs(self, X, W):
-    self.X_ = X = check_array(X)
-    W = check_array(W, accept_sparse=True)
+  def _prepare_pairs(self, pairs, y):
+    pairs, y = check_X_y(pairs, y, accept_sparse=False,
+                                      ensure_2d=False, allow_nd=True)
     # set up prior M
     if self.use_cov:
+      X = np.unique(pairs.reshape(-1, pairs.shape[2]), axis=0)
```

Comment:
This is causing test failures, because the `axis` argument for `np.unique` was added in version 1.13.0 but we test with an earlier version.

Reply:
Yes, indeed, I will change this to something compatible with version 1.12.1

Reply:
I'll go for this solution: https://stackoverflow.com/questions/16970982/find-unique-rows-in-numpy-array/22941699#22941699